xenserver/tests/matchers.py

```diff
@@ -4,70 +4,32 @@
 
 from copy import deepcopy
 
+from testtools.matchers import (
+    Always, ContainsDict, Equals, MatchesDict, MatchesListwise,
+    MatchesSetwise, Never)
 
-class BaseMatcher(object):
-    """
-    The base class for a matcher that does fancy equality checks.
-    """
-    def match(self, other):
-        raise NotImplementedError()
-
-    def __eq__(self, other):
-        return self.match(other)
-
-    def __ne__(self, other):
-        return not self.match(other)
 
-
-class GenericMatcher(BaseMatcher):
+class ExtractValue(object):
     """
-    A matcher that accepts an arbitrary match function.
-    """
-    def __init__(self, match, repr_str):
-        self._match_func = match
-        self._repr_str = repr_str
-
-    def match(self, other):
-        return self._match_func(other)
-
-    def __repr__(self):
-        return self._repr_str
-
-
-NO_VALUE = GenericMatcher(lambda _: False, "<NO VALUE>")
-ANY_VALUE = GenericMatcher(lambda _: True, "<ANY VALUE>")
```

Comment:
I don't really remember writing it, to be honest. It's definitely the kind of thing I'd do, though.